packages/dds/tree/src/shared-tree-core/editManager.ts

```diff
@@ -656,66 +664,73 @@ export class EditManager<
 			0x713 /* Expected change sequence number to exceed the last known minimum sequence number */,
 		);
 
-		assert(
-			sequenceNumber >= // This is ">=", not ">" because changes in the same batch will have the same sequence number
-				(this.sequenceMap.maxKey()?.sequenceNumber ?? minimumPossibleSequenceNumber),
-			0xa64 /* Attempted to sequence change with an outdated sequence number */,
-		);
-
-		const commitsSequenceNumber = this.getBatch(sequenceNumber);
-		const sequenceId: SequenceId =
-			commitsSequenceNumber.length === 0
-				? {
-						sequenceNumber,
-					}
-				: {
-						sequenceNumber,
-						indexInBatch: commitsSequenceNumber.length,
-					};
+		for (const newCommit of newCommits) {
+			assert(
+				sequenceNumber >= // This is ">=", not ">" because changes in the same batch will have the same sequence number
+					(this.sequenceMap.maxKey()?.sequenceNumber ?? minimumPossibleSequenceNumber),
+				0xa64 /* Attempted to sequence change with an outdated sequence number */,
+			);
```

Comment:
Since all the commits in the same batch will have the same sequence number, I think it would make sense to move this assert outside (above) the for loop

Reply:
Is it possible tha the sequneceMap be updated when processing the commits?

Reply:
Yes, but that shouldn't be an issue.